File: chapters/pc02/pc0203/c/pc0203.c
/**
 * pc110203
 * uva10050
 */

#include <stdio.h>

int main(void) {
	int t;

	if (scanf("%d", &t) != 1) {
		return 0;
	}

	for (int case_idx = 0; case_idx < t; case_idx++) {
		int n;
		int p;
		int lost = 0;
		int hartal[101];
		int is_hartal[3651] = {0};

		if (scanf("%d", &n) != 1) {
			return 0;
		}
		if (scanf("%d", &p) != 1) {
			return 0;
		}

		for (int i = 0; i < p; i++) {
			if (scanf("%d", &hartal[i]) != 1) {
				return 0;
			}
		}

		for (int i = 0; i < p; i++) {
			for (int day = hartal[i]; day <= n; day += hartal[i]) {
				int weekday = day % 7;
				if (weekday == 6 || weekday == 0) {
					continue;
				}
				if (!is_hartal[day]) {
					is_hartal[day] = 1;
					lost++;
				}
			}
		}

		printf("%d\n", lost);
	}

	return 0;
}

File: chapters/pc02/pc0203/java/PC0203.java
/*
 * pc110203
 * uva10050
 */

package chapters.pc02.pc0203.java;

import java.util.Scanner;

public class PC0203 {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        if (!sc.hasNextInt()) {
            sc.close();
            return;
        }

        int t = sc.nextInt();

        for (int caseIdx = 0; caseIdx < t; caseIdx++) {
            int n = sc.nextInt();
            int p = sc.nextInt();

            int[] h = new int[p];
            for (int i = 0; i < p; i++) {
                h[i] = sc.nextInt();
            }

            boolean[] isHartal = new boolean[n + 1];
            int lost = 0;

            for (int i = 0; i < p; i++) {
                for (int day = h[i]; day <= n; day += h[i]) {
                    int weekday = day % 7;
                    if (weekday == 6 || weekday == 0) {
                        continue;
                    }

                    if (!isHartal[day]) {
                        isHartal[day] = true;
                        lost++;
                    }
                }
            }

            System.out.println(lost);
        }

        sc.close();
    }
}
